controllers/competition_supervisor/competition_supervisor.py

```diff
@@ -134,6 +137,26 @@ def remove_unused_robots(supervisor: Supervisor) -> None:
         robot.remove()
 
 
+def get_simulation_run_mode(supervisor: Supervisor) -> 'SimulationMode':
+    if supervisor.getDevice("2021a-compatibility") is None:
```

Comment:
I'm a little surprised that this doesn't emit a `TypeError` under `2020b-rev1`, yet it seems that it doesn't. Presumably we're saved by the Supervisor robot not having any devices?
That might be worth a comment, especially as the error (if we did start seeing it) wouldn't be obvious to diagnose.

Reply:
Yes strangely the `TypeError` only seems to occur when the return value is assigned to a variable.
Perhaps something like:
```suggestion
    # webots 2020b is buggy and can raise TypeError when passed a str,
    # however this isn't occuring when the returned value isn't assigned to a variable
    if supervisor.getDevice("2021a-compatibility") is None:
```

Reply:
I seriously doubt that whether or not this is assigned to a variable is going to be affecting whether or not a `TypeError` is thrown. If it is, then something _very_ odd is going on with your Python interpreter. I strongly suspect that there may have been confounding effects when you observed that behaviour.

I've just tested assigning this to a local variable (and then using that in the conditional) and it still works fine.

Unless we know exactly why we're not seeing that error here (as noted I suspect it's because this `Robot` doesn't actually have any devices) I would encourage that we not commit to an explanation.

Reply:
Yeah I can't currently replicate the issue so we should probably just stick to the first line comment

Reply:
see 5cf09f549c0d9819588fe03836b169200903049d